Add "Sanitizing User Input" guide (#11)
bholmesdev/simple-stack: packages/form/README.md

- [StackBlitz playground](https://stackblitz.com/github/bholmesdev/simple-stack/tree/main/examples/form)
- [GitHub](https://github.com/bholmesdev/simple-stack/tree/main/examples/form)

## Sanitizing User Input

> To enhance the security of your application, it's crucial to sanitize user input to prevent Cross-Site Scripting (XSS) attacks. This process ensures that any potentially malicious scripts or code entered by users are neutralized before rendering on the webpage. Here, we'll guide you through using Zod `validation` in combination with the [sanitize-html](https://www.npmjs.com/package/sanitize-html) library for effective sanitization.

Reviewer: ```suggestion
You may need to sanitize user input with rich text content. This is important for any text rendered as HTML to prevent Cross-Site Scripting (XSS) attacks. You can use the [sanitize-html](https://www.npmjs.com/package/sanitize-html) library for this:
```

Simplified the preamble! I structured this based on a common formula I learned in the Astro docs:

1. you may want to do X
2. X is useful when you need to do / prevent Y
3. to start, do...

> 
> ### Install the sanitize-html library
> 
> Before implementing sanitization, install the sanitize-html library along with its type definitions for TypeScript:

Reviewer: ```suggestion
```

We can remove the subheaders and jump straight to the installation instructions!

```bash
npm install --save sanitize-html
npm install --save-dev @types/sanitize-html
```

> ### Use sanitize-html with Zod
> 
> Next, incorporate `sanitize-html` into your Zod validation process by using the `transform` method. This method allows you to apply custom transformations to the data during the validation process.

Reviewer: ```suggestion
Next, call `sanitize-html` from your text validator [with a Zod `transform()`](https://zod.dev/?id=transform):
```

Removed the heading again and simplified the preamble. I've found it's better to link to external docs rather than explaining a different library in your own docs. Saves us from walking people through the Zod library

```diff
+ import sanitizeHtml from "sanitize-html";

const signupForm = createForm({
-  name: z.string(),
+  name: z.string().transform((dirty) => sanitizeHtml(dirty)),
  age: z.number().min(18).optional(),
  newsletterOptIn: z.boolean(),
});
```

> > In the example above, the name field is configured to use the sanitizeHtml transformation. This ensures that any HTML or script tags entered by users will be safely removed, preventing XSS vulnerabilities.

Reviewer: ```suggestion
```

I'm not sure the added explainer is needed. This restates ideas already explained (XSS and Zod transforms)
